### ISSUE TITLE
Label selectors are ignored by the List method of the service.Dashboard interface

### ISSUE THREAD

**Issue by danielpacak** [vmware-tanzu/octant#974]
**What steps did you take and what happened:**

I'm developing this plugin https://github.com/aquasecurity/starboard-octant-plugin, which relies on the `Dashboard` service provided by Plugins API v0.12.1

However, it seems that the implementation of the `List` method ignores label selectors set by my plugin code:

```go
// List lists objects in the dashboard's object store.
func (c *Client) List(ctx context.Context, key store.Key) (*unstructured.UnstructuredList, error) {
	client := c.DashboardConnection.Client()

	keyRequest, err := convertFromKey(key)
	if err != nil {
		return nil, err
	}

	// ... //
}
```

Looking at the implementation of the `convertFromKey` method it ignores the `Selector` property of the `store.Key` argument:

```go
func convertFromKey(in store.Key) (*proto.KeyRequest, error) {
	return &proto.KeyRequest{
		Namespace:  in.Namespace,
		ApiVersion: in.APIVersion,
		Kind:       in.Kind,
		Name:       in.Name,
	}, nil
}
```

Is there any particular reason for ignoring label selectors in this place?

**What did you expect to happen:**

I expect that the `Selector` property is respected by the Plugins API and passed to Kubernetes API through Octant:

```go
unstructuredList, err := r.client.List(ctx, store.Key{
	APIVersion: aquaSecurityAPIVersion,
	Kind:       vulnerabilitiesKind,
	// FIXME This Selector is ignored :( 
	Selector: &labels.Set{
		kube.LabelResourceKind: options.Kind,
		kube.LabelResourceName: options.Name,
	},
})
```

Beyond that, I'd like to know if there's any other way to list Kuberentes customer resource with label selectors to what I'm currently using.

**Anything else you would like to add:**

We heavily rely on labels selectors which are indexed by etcd. So the limitation I've just described might have impact on the performance of the plugin.

Here's the code snippet where I invoke the `List` method on `service.Dashboard` interface: https://github.com/aquasecurity/starboard-octant-plugin/blob/master/pkg/plugin/model/repository.go#L81

**Environment:**

- Octant version (use `octant version`): 
   ```
   Version:  0.13.0 
   Git commit:  d7d0660557c1f2068c35f8f70b005faec1d24811
   Built:  2020-06-01T21:22:38Z
   ```
- Kubernetes version (use `kubectl version`):
   ```
   Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-21T14:51:23Z", GoVersion:"go1.14.3", Compiler:"gc", Platform:"darwin/amd64"}
   Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.6", GitCommit:"d32e40e20d167e103faf894261614c5b45c44198", GitTreeState:"clean", BuildDate:"2020-05-20T13:08:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
   ```
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc):
  `macOS Catalina 10.15.5 (19F101)`

**Comment by GuessWhoSamFoo** [comment]
Does this persist on v0.13.0? This should have been addressed on #908

**Comment by danielpacak** [comment]
> Does this persist on v0.13.0? This should have been addressed on #908

Let me me check. I did run the plugin on Octant 0.13.0, however the plugin itself is compiled with the dependency on `github.com/vmware-tanzu/octant v0.12.1` module

**Comment by danielpacak** [comment]
You're right @GuessWhoSamFoo This issue is gone in 0.13.0 and I do confirm that the plugin is currently using label selectors:

```go
if in.Selector != nil {
	keyRequest.LabelSelector = &wrappers.BytesValue{Value: []byte(in.Selector.String())}
}
```

The only thing was some fun migrating from client-go 0.17.3, which I used so fat, to v0.19.0-alpha.3 required by Octant 0.13.0
